src/scala/com/github/johnynek/bazel_deps/GradleResolver.scala

```diff
@@ -0,0 +1,275 @@
+package com.github.johnynek.bazel_deps
+
+import cats.MonadError
+import coursier.util.Task
+import io.circe.jawn.JawnParser
+import cats.implicits._
+
+import java.io.File
+import java.nio.file.Path
+import scala.collection.immutable.SortedMap
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.Duration
+import scala.util.{Failure, Success, Try}
+
+import cats.implicits._
+
+class GradleResolver(
+    servers: List[DependencyServer],
+    ec: ExecutionContext,
+    runTimeout: Duration,
+    gradleTpe: ResolverType.Gradle,
+    cachePath: Path
+) extends Resolver[Task] {
+  private[this] lazy val coursierResolver =
+    new CoursierResolver(servers, ec, runTimeout, cachePath)
+
+  implicit def resolverMonad: MonadError[Task, Throwable] =
+    coursierResolver.resolverMonad
+
+  def getShas(
+      m: List[MavenCoordinate]
+  ): Task[SortedMap[MavenCoordinate, ResolvedShasValue]] =
+    coursierResolver.getShas(m)
+
+  private def loadLockFiles(
+      lockFiles: List[String]
+  ): Try[List[GradleLockFile]] =
+    lockFiles.traverse { next =>
+      (Model.readFile(new File(next)) match {
+        case Success(str) => Success(str)
+        case Failure(err) =>
+          Failure(new Exception(s"Failed to read ${next}", err))
+      })
+      .flatMap { content =>
+        Decoders.decodeGradleLockFile(new JawnParser, content) match {
+          case Right(m) => Success(m)
+          case Left(err) =>
+            Failure(new Exception(s"Failed to parse ${next}", err))
+        }
+      }
+    }
+
+  private def mergeLockFiles(
+      lockFiles: List[GradleLockFile]
+  ): Try[GradleLockFile] =
+    lockFiles.foldM(GradleLockFile.empty) { (s, n) =>
+      TryMerge.tryMerge(None, s, n)
+    }
+
+  // Gradle has compile/runtime/test sepearate classpaths
+  // we just want one, so we merge them all
+  private def collapseDepTypes(
+      lockFile: GradleLockFile
+  ): Try[Map[String, GradleLockDependency]] =
+    List(
+      lockFile.compileClasspath,
+      lockFile.annotationProcessor,
+      lockFile.resolutionRules,
+      lockFile.runtimeClasspath,
+      lockFile.testCompileClasspath,
+      lockFile.testRuntimeClasspath
+    )
+      .map(_.getOrElse(Map()))
+      .foldLeft(Try(Map[String, GradleLockDependency]())) { case (p, n) =>
+        p.flatMap { s => TryMerge.tryMerge(None, s, n) }
+      }
```

Comment:
Does this mean that the transitive dependencies of `test` dependencies could override the `compile` configuration? This could potentially cause prod differences between Gradle and Bazel.

Reply:
yes, that's what it means. It's intentional. We can discuss later. We want a single version of all the jars. In fact, gradle having a different version for each project main and test has caused prod issues: we know these jars will be combined at runtime, so ignoring that and letting them transitively get different dependencies at compile and test time is asking for trouble.